experiment_buddy/experiment_buddy.py

```diff
@@ -232,57 +240,49 @@ def _setup_tb(logdir):
     return tensorboardX.SummaryWriter(logdir=logdir)
 
 
-def _open_ssh_session(hostname: str) -> fabric.Connection:
+def open_ssh_session(hostname: str) -> fabric.Connection:
     """ TODO: move this to utils.py or to a class (better)
         TODO add time-out for unknown host
      """
-
     try:
         ssh_session = fabric.Connection(host=hostname, connect_timeout=10, forward_agent=True)
         ssh_session.run("")
+        return ssh_session
```

Comment:
return goes out of the try, in general you want to have the least number of statements in the try, to avoid capturing the exception from the wrong statement.
also don't change the code unless there is a functional reason to reduce the line changes